packages/react-core/src/components/Truncate/__tests__/Truncate.test.tsx

```diff
@@ -1,15 +1,117 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 import { Truncate } from '../Truncate';
 
+jest.mock('../../Tooltip', () => ({
+  Tooltip: ({ content, tooltipPosition, children, ...props }) => (
+    <>
+      <div {...props}>{children}</div>
+    </>
+  )
+}));
+
+test('renders with class pf-c-truncate', () => {
+  render(
+    <Truncate
+      content={''}
+      aria-label='test-id'
+    />
+  );
+
+  const test = screen.getByLabelText('test-id');
+
+  expect(test).toHaveClass('pf-c-truncate');
+});
+
+test('renders with custom class name passed via prop', () => {
+  render(
+    <Truncate
+      content={''}
+      aria-label='test-id'
+      className='custom-classname'
+    />
+  );
+
+  const test = screen.getByLabelText('test-id');
+
+  expect(test).toHaveClass('custom-classname');
+});
+
+test('renders truncate with empty content', () => {
+  render(
+    <Truncate
+      content={''}
+      aria-label='test-id'
+    />
+  );
+
+  const test = screen.getByLabelText('test-id');
+
+  expect(test).toHaveTextContent('');
+});
+
+test('renders truncate with content', () => {
+  render(
+    <Truncate
+      content={'Test'}
+      aria-label='test-id'
+    />
+  );
+
+  const test = screen.getByLabelText('test-id');
+
+  expect(test).toHaveTextContent('Test');
+});
+
+test('renders pf-c-truncate__start when trailingNumChars prop passed', () => {
+  render(
+    <Truncate
+      content={'Testing truncate content'}
+      trailingNumChars={3}
+      position='middle'
+    />
+  );
+
+  const start = screen.getByText('Testing truncate cont');
+
+  expect(start).toHaveClass('pf-c-truncate__start');
+});
+
+test('renders pf-c-truncate__end when trailingNumChars prop passed', () => {
+  render(
+    <Truncate
+      content={'Testing truncate content'}
+      trailingNumChars={3}
+      position='middle'
+    />
+  );
+
+  const end = screen.getByText('ent');
+
+  expect(end).toHaveClass('pf-c-truncate__end');
+});
+
+test('only renders pf-c-truncate__start if truncate position is the end', () => {
+  render(
+    <Truncate
+      content={'Testing truncate content'}
+      position='end'
+    />
+  );
+
+  const start = screen.getByText('Testing truncate content');
+
+  expect(start).toHaveClass('pf-c-truncate__start');
+});
```

Comment:
Since position=end is the default value I think this test should actually omit the position prop so that we have a test about the default behavior here.